exonum/src/helpers/fabric/details.rs

```diff
@@ -392,6 +406,47 @@ impl GenerateNodeConfig {
 
         (external_address, listen_address)
     }
+
+    /// * `key` - expects EXONUM_SERVICE_PASS or EXONUM_CONSENSUS_PASS.
+    fn get_passphrase(context: &Context, key: &str) -> Vec<u8> {
+        context
+            .get_flag_occurrences(NO_PASSWORD)
+            .map(|_| "".to_string())
+            .or_else(|| env::var(key).ok())
+            .or_else(|| {
+                let stdin = io::stdin();
+                let stdin_locked = stdin.lock();
+                Self::prompt_passphrase(stdin_locked, io::stdout()).ok()
+            }).unwrap()
+            .into_bytes()
+    }
+
+    fn prompt_passphrase<R: BufRead, W: Write>(mut reader: R, mut writer: W) -> io::Result<String> {
+        loop {
+            write!(&mut writer, "Enter passphrase (empty for no passphrase): ").unwrap();
+            let password = rpassword::read_password_with_reader(Some(&mut reader))?;
+            write!(&mut writer, "Enter same passphrase again: ").unwrap();
+            if password == rpassword::read_password_with_reader(Some(&mut reader))? {
+                return Ok(password);
+            }
+            writeln!(&mut writer, "Passphrases do not match.  Try again.").unwrap();
+        }
+    }
+
+    fn create_secret_key_file(
+        secret_key_path: impl AsRef<Path>,
+        passphrase: impl AsRef<[u8]>,
+    ) -> PublicKey {
+        let secret_key_path = secret_key_path.as_ref();
+        if secret_key_path.exists() {
+            panic!("{}: File exists", secret_key_path.to_string_lossy(),);
```

Comment:
These panics and unwraps make me cringe a bit. We can be more nice and return a Result from this function and _then_ panic in one place if we wish to. A function like this should not make decisions on error handling.

I'm not against leaving it as is (because of gazillion of other unwraps that are already present in this module), but following the boy scout rule is good for one's karma.